pkg/apis/apps/v1beta1/types.go

```diff
@@ -25,6 +25,10 @@ import (
 const (
 	// StatefulSetInitAnnotation if present, and set to false, indicates that a Pod's readiness should be ignored.
 	StatefulSetInitAnnotation = "pod.alpha.kubernetes.io/initialized"
+	// StatefulsetBurstAnnotation instructs the stateful set controller to not wait when replacing pods from the set,
+	// and that it may create pods out of order. If enabling this annotation, ensure that your set is capable of
+	// forming a safe quorum even if pod creation is out of order. Accepts "true".
+	StatefulSetBurstAnnotation = "statefulset.alpha.kubernetes.io/burst"
```

Comment:
Initial proposal in the issue was for an annotation, @bgrant0607 seems to favor just adding the field if we want to evolve it from an annotation to a field. I prefer the field primarily for validation integration and API clarity, but, as the annotation is functionally equivalent, if you want to go in that direction I'm on board.